### DISCUSSION TITLE
random border color for each pdf page

### DISCUSSION THREAD

**Discussion by srini8080** [ImageMagick/ImageMagick#4799] [Development]
Is there a way we can create a random bordercolor for each page of a PDF file?

Thanks,

**Comment by fmw42** [comment]
For what version of ImageMagick and what platform?  Do you mean random values all around or each page has a constant  color border but randomly different from the other pages?

**Comment by srini8080** [comment]
IM v7 Centos 8
each page has a constant color border but randomly different from the other pages?

Thanks,
    On Thursday, February 3, 2022, 08:26:51 AM GMT+5:30, Fred Weinhaus ***@***.***> wrote:  
 
 


For what version of ImageMagick and what platform? Do you mean random values all around or each page has a constant color border but randomly different from the other pages?

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fmw42** [comment]
Do you have a PDF to use? Please provide.

**Comment by fmw42** [comment]
What output format? If PDF, then know that ImageMagick rasterizes each page, so that any text will not be editable.

**Comment by srini8080** [comment]
yes. PDF.
Not editable text is fine.
Thanks
    On Thursday, February 3, 2022, 10:11:47 AM GMT+5:30, Fred Weinhaus ***@***.***> wrote:  
 
 


What output format? If PDF, then know that ImageMagick rasterizes each page, so that any text will not be editable.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by fmw42** [comment]
You can do it using sub shell processing on Unix-like systems.  Here is a 3 page PDF created from 3 images. (I asked you for a pdf, but you did not provide one).

[image.pdf](https://github.com/ImageMagick/ImageMagick/files/7996694/image.pdf)


```
(
for ((i=0; i<3; i++)); do
magick image.pdf[$i] -bordercolor "hsv(%[fx:360*rand()],%[fx:255*rand()],%[fx:255*rand()])" -border 10 miff:-
done 
) | magick - x.pdf

```

[x.pdf](https://github.com/ImageMagick/ImageMagick/files/7996696/x.pdf)


(I tried using %[fx:s] to get the image sequence number, but could not make that work in a single command line.)

**Reply by srini8080** [reply]
Thanks, that worked.
